### ISSUE TITLE
problem of reschedule:on-node-failure with Consul

### ISSUE THREAD

**Issue by shenshouer** [docker/swarm#1837]
I have 4 machine such as:

* swarm-01 consul && swarm master
* swarm-02 swarm master
* swarm-03 swarm node
* swarm-04 swarm node

version:
* coreos 955.0.0
* docker 1.10.0
* swarm v1.1.0 or v1.1.1
* compose 1.6.0

```
core@swarm-01 ~ $ docker -H :4000 info
Containers: 1
 Running: 0
 Paused: 0
 Stopped: 1
Images: 1
Server Version: swarm/1.1.0
Role: replica
Primary:
Strategy: spread
Filters: health, port, dependency, affinity, constraint
Nodes: 1
 swarm-03: 172.18.19.103:2375
  └ Status: Healthy
  └ Containers: 1
  └ Reserved CPUs: 0 / 2
  └ Reserved Memory: 0 B / 2.056 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=4.4.1-coreos, operatingsystem=CoreOS 955.0.0, storagedriver=overlay
  └ Error: (none)
  └ UpdatedAt: 2016-02-17T11:23:44Z
Plugins:
 Volume:
 Network:
Kernel Version: 4.4.1-coreos
Operating System: linux
Architecture: amd64
CPUs: 2
Total Memory: 2.056 GiB
Name: swarm-01
Experimental: true
core@swarm-01 ~ $ swarm l consul://localhost:8500
INFO[0000] Initializing discovery without TLS
172.18.19.103:2375
172.18.19.104:2375
```

### I run one container use docker command:
`core@swarm-01 ~ $ DOCKER_HOST=127.0.0.1:4000 docker run -P -d --name=web -e reschedule:on-node-failure nginx`

found it run at the swarm-04:

```
core@swarm-01 ~ $ DOCKER_HOST=127.0.0.1:4000 docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                                                       NAMES
d4c6cb82bf00        nginx               "nginx -g 'daemon off"   3 seconds ago       Up 2 seconds        172.18.19.104:32769->80/tcp, 172.18.19.104:32768->443/tcp   swarm-04/web
```
Then I stop the service of swarm node on node swarm-04. It can reschedule to swarm-03, This was expected:

```
core@swarm-01 ~ $ docker run --rm swarm l consul://172.18.19.101:8500
time="2016-02-18T03:12:48Z" level=info msg="Initializing discovery without TLS"
172.18.19.103:2375
core@swarm-01 ~ $ DOCKER_HOST=127.0.0.1:4000 docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS               NAMES
6ef77b03593c        nginx               "nginx -g 'daemon off"   15 seconds ago      Up 14 seconds       80/tcp, 443/tcp     swarm-03/web
```
If restart swarm node service on node swarm-04, but here comes the question, there is two containers on cluster with the same name:

````
core@swarm-01 ~ $ docker run --rm swarm l consul://172.18.19.101:8500
time="2016-02-18T03:15:02Z" level=info msg="Initializing discovery without TLS"
172.18.19.103:2375
172.18.19.104:2375
core@swarm-01 ~ $ DOCKER_HOST=127.0.0.1:4000 docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS               NAMES
6ef77b03593c        nginx               "nginx -g 'daemon off"   2 minutes ago       Up 2 minutes        80/tcp, 443/tcp     swarm-03/web
d314a91b8c07        nginx               "nginx -g 'daemon off"   3 minutes ago       Up 3 minutes        80/tcp, 443/tcp     swarm-04/web
````

### Then I use docker-compose to do this.
the content of docker-compose as:

```
web:
  image: nginx
  restart: always
  labels:
    com.docker.swarm.reschedule-policy: "on-node-failure"
```
or

```
web:
  image: nginx
  restart: always
  environment:
    reschedule: on-node-failure
```

But the container didnot reschedule at all.

According to the above content, I have two questions:

* Does swarm keep the count of container running ?
* Does compose support the reschedule with swarm ?

**Comment by shenshouer** [comment]
Got it working in `docker-compose` like so:
```
web:
  image: nginx
  restart: always
#  environment:
#    reschedule: on-node-failure
  labels:
#    com.docker.swarm.reschedule-policy: "[\"on-node-failure\"]"
    com.docker.swarm.reschedule-policies: "[\"on-node-failure\"]"
```

**Comment by vieux** [comment]
@shenshouer the `environment` part will be fixed in compose 1.6.1

For the rest I added a test here: https://github.com/docker/swarm/pull/1856 to make sure it works

**Comment by vieux** [comment]
This is fixed in compose 1.6.2 see https://github.com/docker/swarm/pull/1881
